annotationframeworkclient/materializationengine.py

```diff
@@ -596,25 +596,50 @@ def _update_rootids(self, df, timestamp):
         time_d = {}
         starttime=time.time()
         sv_columns = [c for c in df.columns if c.endswith('supervoxel_id')]
+        all_root_ids = []
+        
+        # go through the columns and collect all the root_ids to check
+        # to see if they need updating
         for sv_col in sv_columns:
             root_id_col = sv_col[:-len('supervoxel_id')] + 'root_id'
-            svids = df[sv_col].values
-            root_ids = df[root_id_col].values.copy()
-
-            uniq_root_ids = np.unique(root_ids)
-            is_latest_root = self.cg_client.is_latest_roots(uniq_root_ids, timestamp=timestamp)
-            is_latest_root = np.isin(root_ids, uniq_root_ids[is_latest_root])
-
-            time_d[f'is_latest_root {sv_col}']=time.time()-starttime
-            starttime=time.time()
+            all_root_ids.append(df[root_id_col].values.copy())
+        
+        uniq_root_ids = np.unique(np.concatenate(all_root_ids))
+        is_latest_root = self.cg_client.is_latest_roots(uniq_root_ids, timestamp=timestamp)
+        latest_root_ids = uniq_root_ids[is_latest_root]
+        time_d[f'is_latest_root']=time.time()-starttime
```

Comment:
Akhilesh uses a nice timing package that might be worth checking out. E.g. https://github.com/seung-lab/PyChunkedGraph/blob/akhilesh-unit-tests/pychunkedgraph/io/edges.py#L67

Reply:
adopted a variant of this, but using logging instead of print